syntax = "proto3";

import "map_renderer.proto";
import "transport_router.proto";
import "graph.proto";

package transport_base;


message Stop {	
	double geo_lat = 1;
	double geo_lng = 2;	
	string name = 3;
}

message Bus {
	uint32 end_stop_ind = 1;
	string name = 2;
	repeated uint32 stop_index = 3;
	bool round_trip = 4;	
}

message Dist {
	uint32 from = 1;
	uint32 to = 2;
	uint32 distance = 3;
}

message TransportCatalogue {
	repeated Stop stops = 1;
	repeated Bus buses = 2;
	repeated Dist distances = 3;
	RenderSettings render_settings = 4;
	TransportRouter transport_router = 5;
	Router router = 6;
	Graph graph = 7;
}
